### PR TITLE
initial Junit5 work

### PR DESCRIPTION
**Pull request by tomnis**
- migrate almost all our tests to jupiter engine
- new meta-annotations `UnitTest` and `IntegTest` are the composition of `@Test` and a corresponding `@Tag` (similar to junit4 `@Category`)
- our tests that rely on junit4 features (Rule, Runner, etc) are executed with junit vintage engine
- both engines can exist on the classpath without conflict
- next steps: extension points for measuring tests

### CODE REVIEW

File: src/test/scala/com/workday/telemetron/spec/HasRandomTestId.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.workday.telemetron.spec
+
+import java.util.UUID
+
+/**
+  * Created by tomas.mccandless on 6/16/20.
+  */
+trait HasRandomTestId {
+
+  def randomTestId(): String = this.getClass.getCanonicalName + "." + UUID.randomUUID().toString
```
Comment (tomnis) [inline src/test/scala/com/workday/telemetron/spec/HasRandomTestId.scala:10]: used only for testing where we need an arbitrary Id

File: gradle.properties [hunk 2]
```diff
@@ -1,9 +1,10 @@
 # scala cross compilation
 # by default, gradle invocations will use only this version. useful for ide invocations.
-defaultScalaVersions=2.12.11
```
Comment (tsoppet) [inline gradle.properties:3]: was downgrading done on purpose?

File: gradle.properties [hunk 3]
```diff
@@ -1,9 +1,10 @@
...
+defaultScalaVersions=2.12.10
 # if we need to cross-compile and rerun the entire task graph (like in a ci build or publish), run with -PallScalaVersions
 # see https://github.com/ADTRAN/gradle-scala-multiversion-plugin#examples
-scalaVersions=2.12.11
+# we are stuck on 2.12.10 for now. see https://github.com/scalacenter/scalafix/issues/998
```
Comment (tomnis) [inline gradle.properties:6]: i will add some more detail about this

File: src/test/scala/com/workday/warp/common/PropertyInspectorSpec.scala [hunk 4]
```diff
@@ -1,9 +1,8 @@
 package com.workday.warp.common
 
-import com.workday.warp.common.category.UnitTest
 import com.workday.warp.common.spec.WarpJUnitSpec
-import org.junit.experimental.categories.Category
-import org.junit.{Ignore, Test}
+import com.workday.warp.junit.UnitTest
+import org.junit.jupiter.api.Disabled
```
Comment (rooeque) [inline src/test/scala/com/workday/warp/common/PropertyInspectorSpec.scala:5]: 👍

**Pull request merged**